code/modules/reagents/reagent_containers.dm

```diff
@@ -328,6 +328,9 @@
 
 	// Ensure we don't splash beakers and similar containers.
 	if(!target.is_open_container() && istype(target, /obj/item/reagent_containers))
+		if(istype(target, /obj/item/reagent_containers/glass/bottle/syrup))
+			var/obj/item/reagent_containers/glass/bottle/syrup/dispenser = target
+			return dispenser.standard_pour_into(user, src)
```

Comment:
this should be a variable or a flag, not hardchecked

Reply:
Added an atom flag, though it still checks for a reagent container type due to it needing to be one to use standard_pour_into()